pkg/scd/operations_handler.go

```diff
@@ -30,29 +31,42 @@ func (a *Server) DeleteOperationReference(ctx context.Context, req *scdpb.Delete
 		return nil, dsserr.PermissionDenied("missing owner from context")
 	}
 
-	// Delete Operation in Store
-	op, subs, err := a.Store.DeleteOperation(ctx, id, owner)
-	if err != nil {
-		return nil, err
-	}
-	if op == nil {
-		return nil, dsserr.Internal(fmt.Sprintf("DeleteOperation returned no Operation for ID: %s", id))
-	}
-	if subs == nil {
-		return nil, dsserr.Internal(fmt.Sprintf("DeleteOperation returned nil Subscriptions for ID: %s", id))
+	var response *scdpb.ChangeOperationReferenceResponse
+	action := func(store scdstore.Store) (err error) {
```

Comment:
this doesn't need to be done in a txn

Reply:
When we simplify the Store APIs to do just simple CRUD (rather than packaging the application logic in the Store layer as is the case currently), changing an Operation Reference will involve retrieving the current version of that Operation Reference, making sure that all necessary preconditions (like owner, version) are satisfied, and then write the modified version back into the database.  The read must be performed in a transaction with the write.

Reply:
actually it can be done in a single sql call, like how we do it in RID :)

Get's as well don't need to be run a txn. Giving us the flexibility to not run queries in a txn will help scale the DSS a lot as it will alleviate contention

Reply:
Performing this action in a single SQL call embeds application logic in the Store layer that I think we generally want to avoid in favor of simple CRUD operations.  In this way, I think that RID's current approach of using two separate calls to [read](https://github.com/interuss/dss/blob/6ae81085411f569adc64dd2a79e525edd2162ce8/pkg/rid/application/isa.go#L87) and [modify](https://github.com/interuss/dss/blob/6ae81085411f569adc64dd2a79e525edd2162ce8/pkg/rid/application/isa.go#L136) an ISA is better than embedding application logic (checking and incrementing version, checking owner) in the Repo layer.

However, there appears to be a latent bug in [app.InsertISA](https://github.com/interuss/dss/blob/6ae81085411f569adc64dd2a79e525edd2162ce8/pkg/rid/application/isa.go#L86) where two racing calls could both pass their app-level version checks, but then one of the two racing modifications would hit the Repo-level version check and return a 500 rather than the desired 409 "old version" because the error returned by the query is returned directly to the gateway.

One way to fix this could be to eliminate the version check in app.InsertISA, instead relying on the Repo layer to perform the version check and return an appropriate error.  However, if we did that, we would definitely need to [document](https://github.com/interuss/dss/blob/0f305734354c8e8637ec65f75527b6688fe75bee/pkg/rid/repos/isa.go#L20) the required behavior of all Repo implementations (validating versions, how exactly to increment the version including the algorithm to generate the version, validate owner) as well as documenting exactly which intended errors might be returned (like the one that should trigger the 409).  All implementations, including mocks, would need to implement this logic.

A different way to fix the problem would be to simply place the initial GetISA in the same transaction as the later InsertISA.  In that case, there would be no need to perform the check in the query, and no need to document the required application logic to be implemented in all Repo implementations.  This option seems like a much simpler route to me.

Reply:
We've already seen some issues due to contention. I think improving the user experience is more important than improving the developer experience. 

In regard to ensuring we return the correct version, we could check for a sql.ErrNoRows error if it is an update, and return a version mismatch (like we do for delete)